libkbfs/mdserver_disk.go

```diff
@@ -5,97 +5,66 @@
 package libkbfs
 
 import (
-	"bytes"
-	"encoding/binary"
 	"errors"
-	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"reflect"
 	"sync"
-	"time"
 
 	"github.com/keybase/client/go/logger"
 	keybase1 "github.com/keybase/client/go/protocol"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/storage"
-	"github.com/syndtr/goleveldb/leveldb/util"
 	"golang.org/x/net/context"
 )
 
-// TODO: Convert this to use flat files and a journal.
+type mdServerDiskShared struct {
+	dirPath string
 
-// MDServerDisk just stores blocks in local leveldb instances.
-type MDServerDisk struct {
-	config   Config
-	handleDb *leveldb.DB // folder handle                  -> folderId
-	mdDb     *leveldb.DB // folderId+[branchId]+[revision] -> mdBlockLocal
-	branchDb *leveldb.DB // folderId+deviceKID             -> branchId
-	log      logger.Logger
-
-	locksMutex *sync.Mutex
-	locksDb    *leveldb.DB // folderId -> deviceKID
+	// Protects handleDb and tlfStorage. After Shutdown() is
+	// called, both handleDb and tlfStorage are nil.
+	lock       sync.RWMutex
+	handleDb   *leveldb.DB // folder handle -> folderId
+	tlfStorage map[TlfID]*mdServerTlfStorage
 
 	updateManager *mdServerLocalUpdateManager
 
-	shutdownLock *sync.RWMutex
-	shutdown     *bool
 	shutdownFunc func(logger.Logger)
 }
 
-var _ mdServerLocal = (*MDServerDisk)(nil)
+// MDServerDisk stores all info on disk.
+type MDServerDisk struct {
+	codec  Codec
```

Comment:
Done, although I'd like to keep them for the per-TLF storage, since those only store codec and cryptoPure, which shouldn't change at all.